File: DS & Algo in python/bfs.py
import collections
# build a graph
graph = {'A': ['B', 'C'],
             'B': ['A', 'D', 'E'],
         'C': ['A', 'F'],
         'D': ['B'],
         'E': ['B', 'F'],
         'F': ['C', 'E']}

# breath first search to traverse graph
def bfs(graph, start):
    visited = []
    queue = [start]
    while queue:
        vertex = queue.pop(0)
        if vertex not in visited:
            visited.append(vertex)
            queue.extend(graph[vertex]) # adds all the vertices connected to vertex to the queue

    return visited

def bfs2(self, node): # BFS 
    if not node:
        return None
    
    clone = {node : Node(val=node.val,neighbors = [])}
    queue = [node]
    while queue:
        print(queue)
        child = queue.pop(0)
            
        # visit neighbors
        for neighbor in child.neighbors:
                        
            if neighbor not in clone: # clone neighbor node and add that to the clone
                queue.append(neighbor)
                clone[neighbor] = Node(neighbor.val,[])
                
            
            clone[child].neighbors.append(clone[neighbor])
            
                
    return clone[node]

def dfs_iterative(self, node):
    if not node:
        return node
    root = Node(node.label)
    stack = [node]
    visit = {}
    visit[node.label] = root
    while stack:
        top = stack.pop()
    
        for n in top.neighbors:
            if n.label not in visit:
                stack.append(n)
                visit[n.label] = Node(n.label)
            visit[top.label].neighbors.append(visit[n.label])
    
    return root

File: Datastructure and Algo in Golang/BFS.go
package main

// create a breath first search algorithm
func BFS(graph [][]int, start int) []int {
	// create a queue to store the path
	queue := make([]int, 0)

	// mark the start node as visited
	visited := make([]bool, len(graph))

	// push the start node into the queue
	queue = append(queue, start)

	// mark the start node as visited
	visited[start] = true

	// create a array to store the path
	path := make([]int, len(graph))

	// while the queue is not empty
	for _, v := range queue {
		// visit the adjacent node
		for _, w := range graph[v] {
			// if the adjacent node is not visited
			if !visited[w] {
				// mark the adjacent node as visited
				visited[w] = true
				// add the adjacent node to the queue
				queue = append(queue, w)
				// add the adjacent node to the path
				path[w] = v
			}
		}
	}

	return path
}
